### PR TITLE
theme: add dark mode

### PR DESCRIPTION
**Pull request by douglastofoli**
# Description
Added a toggle button for change css theme to Dark Mode. The toggle stores a state for the currently theme and also a localStorage is set to remember the user decision for the next access on the website.

# TODO

- [x] Fix delay to load theme on reload page
- [x] Fix theme when change from CFP to another page
- [x] Adjust on the css from theme switcher. Missing responsive.

### CODE REVIEW

File: src/components/Header.svelte [hunk 1]
```diff
@@ -14,6 +15,7 @@
     <a href={`${base}/#about`}><Localized id="header--about" /></a>
     <a href={`${base}/cfp`}><Localized id="header--cfp" /></a>
     <LanguageSwitcher />
+    <ThemeSwitcher />
```
Comment (macabeus) [inline src/components/Header.svelte:18]: The header has too many elements now, and the `Home` item is overlapped when in a small resolution (e.g., on an iPhone).
We should replace `Home / Sobre / CFP` with a dropdown when using a small resolution.
telegram-cloud-photo-size-1-4963464739736890066-y

Comment (macabeus) [inline src/components/Header.svelte:18]: The header has too many elements now, and the `Home` item is overlapped when in a small resolution (e.g., on an iPhone).
We should replace `Home / Sobre / CFP` with a dropdown when using a small resolution.
telegram-cloud-photo-size-1-4963464739736890066-y

File: static/themes/dark.css [hunk 2]
```diff
@@ -0,0 +1,76 @@
...
+.tweet {
+  background-color: rgb(24, 26, 27);
+}
+
+/* Images */
+
+/* Cumbuca logo */
+.content a:nth-child(4) > img {
+  filter: brightness(0) !important;
+}
```
Comment (macabeus) [inline static/themes/dark.css:76]: We should not override images from a theme CSS file.
Also, we are going to remove the Cumbuca logo soon: https://github.com/gambiconf/gambiconf.github.io/pull/20
```suggestion
```

File: static/themes/dark.css [hunk 3]
```diff
@@ -0,0 +1,76 @@
+::selection {
+  background-color: #004daa !important;
```
Comment (macabeus) [inline static/themes/dark.css:2]: There are too many `!important` here.
It should not be necessary. We should not have a `dark.css` overridden the `app.css` styles.

Instead, we should have the styles defined on `light.css`, `dark.css`, and on `app.css` only the common styles for both themes that don't depend on the color schema (for instance, the font sizes).
Then, we won't need to use `!important`.

File: static/themes/dark.css [hunk 4]
```diff
@@ -0,0 +1,76 @@
...
+
+div.wrapper-content {
+  background-image: linear-gradient(rgba(24, 26, 27, 0.5), rgba(24, 26, 27, 0.5)),
+    url("/mambi-icon-oulined.png") !important;
+}
+
+.window {
+  border-color: rgb(129, 120, 106) !important;
+  background-color: rgb(24, 26, 27) !important;
+}
```
Comment (macabeus) [inline static/themes/dark.css:55]: This style is too specific for a component.
We should define the style for a component within the component file. The same of the other components here.

File: src/components/ThemeSwitcher.svelte [hunk 5]
```diff
@@ -0,0 +1,75 @@
...
+      return current === 'light' ? 'dark' : 'light'
+    })
+
+    theme.subscribe((value) => {
+      localStorage.setItem('theme', value)
+    })
+  }
+
+  onMount(() => {
+    let currentTheme = localStorage.getItem('theme') ?? 'light'
```
Comment (macabeus) [inline src/components/ThemeSwitcher.svelte:21]: If there is no `localStorage.getItem('theme')`, we can also fallback to the user agent preferences: https://stackoverflow.com/a/57795495/3440266

File: src/routes/+layout.svelte [hunk 6]
```diff
@@ -4,10 +4,14 @@
   import Seo from "../components/Seo.svelte"
   import Header from "../components/Header.svelte"
   import LocalizationProvider from "../providers/LocalizationProvider.svelte"
+  import { theme } from '../store/theme'
   import "../app.css"
 </script>
 
 <svelte:head>
+  <meta name="color-scheme" content={$theme === 'light' ? 'light' : $theme} />
```
Comment (macabeus) [inline src/routes/+layout.svelte:12]: Is it not the same as that?
```suggestion
  <meta name="color-scheme" content={$theme} />
```

File: src/store/theme.ts [hunk 7]
```diff
@@ -0,0 +1,3 @@
+import { writable } from 'svelte/store';
+
+export const theme = writable('light');
```
Comment (macabeus) [inline src/store/theme.ts:3]: We can be type safer.
```suggestion
type Theme = 'light' | 'dark'

export const theme = writable<Theme>('light')
```

File: static/themes/dark.css (hunk 2 above)
Comment (douglastofoli) [reply static/themes/dark.css]: Cumbuca logo style was removed

File: src/routes/+layout.svelte (hunk 6 above)
Comment (douglastofoli) [reply src/routes/+layout.svelte]: Yes, it's the same. I removed the condition.

File: src/components/Header.svelte (hunk 1 above)
Comment (macabeus) [reply src/components/Header.svelte:18]: Okay, we can postpone it

File: l10n/pt.ftl [hunk 8]
```diff
@@ -111,3 +111,7 @@ cfp--section-contacts = Contatos
 cfp--field-social-medias = Medias sociais (isso será público)
 cfp--field-email = E-Mail para o feedback (não será publicado)
 cfp--submit = Submeter
+
+# ThemeSwitch
+theme--light = Trocar para tema escuro
+theme--dark = Trocar para tema claro
```
Comment (macabeus) [inline l10n/pt.ftl:117]:
```suggestion
theme--light = Trocar para dark mode
theme--dark = Trocar para light mode
```

File: l10n/en.ftl [hunk 9]
```diff
@@ -111,3 +111,7 @@ cfp--section-contacts = Contacts
 cfp--field-social-medias = Social Medias (it'll be public)
 cfp--field-email = E-Mail for the feedback (we won't share it)
 cfp--submit = Submit
+
+# ThemeSwitch
+theme--light = Switch to dark theme
+theme--dark = Switch to light theme
```
Comment (macabeus) [inline l10n/en.ftl:117]:
```suggestion
theme--light = Switch to dark mode
theme--dark = Switch to light mode
```

File: src/components/Tweet.svelte
Comment (macabeus) [inline src/components/Tweet.svelte:1]: It's missing to improve two things here:
- the inner circle should be transparent:
image
- the three dots should have a better color when in dark mode
image

**Review by macabeus — COMMENTED** [review]
It's almost done 🎉 
Just two small missing things

**Comment by macabeus** [comment]
Thanks for the great work!

**Pull request merged**